server/db/seed.js

```diff
@@ -0,0 +1,179 @@
+  }
+];
+const contactListParams = {
+  name: 'Contact List 1'
+};
+let campaignScriptId;
+
+function createUser(params) {
+  return new User(params).save()
+  .then().catch(err => console.log(err));
+}
+
+function createQuestion(params) {
+  return questionsService.saveNewQuestion(params)
+    .then().catch(err => console.log(err));
+}
+
+function createScriptQuestion(params) {
+  return scriptsService.addQuestionToScript(params)
+    .then().catch(err => console.log(err));
+}
+
+function createContact(params) {
+  return contactsService.saveNewContact(params)
+    .then().catch(err => console.log(err));
+}
+
+function createContactListContact(params) {
+  return contactListsService.addContactToContactList(params)
+  .then().catch(err => console.log(err));
+}
+
+function generatePromiseActions(paramsArray, createFunction) {
+  const callArray = [];
+
+  paramsArray.forEach((paramsObject, index) => {
+    callArray.push(createFunction(paramsArray[index]));
+  });
+
+  return callArray;
+}
+
+Promise.all(generatePromiseActions(userParams, createUser));
+Promise.all(generatePromiseActions(questionParams, createQuestion))
+  .then((questions) => {
+    scriptsService.saveNewScript(scriptParams)
+      .then((script) => {
+        campaignScriptId = script.attributes.id;
+        // console.log(questions);
+        const id = campaignScriptId;
+        const questionIds = questions.map(question => question.attributes.id);
+        const scriptQuestionParams = [];
+        questionIds.forEach((questionId, index) => {
+          scriptQuestionParams.push({
+            id,
+            question_id: questionId,
+            sequence_number: index + 1
+          });
+        });
+
+        Promise.all(generatePromiseActions(scriptQuestionParams, createScriptQuestion))
+          .then().catch(err => console.log(err));
+      }).catch(err => console.log(err));
+  }).catch(err => console.log(err));
+
+contactListsService.saveNewContactList(contactListParams)
+  .then((contactList) => {
+    Promise.all(generatePromiseActions(contactParams, createContact))
+      .then((contacts) => {
+        const contactIds = contacts.map(contact => contact.attributes.id);
+        const contactListContactParams = [];
+
+        contactIds.forEach((contactId) => {
+          contactListContactParams.push({
+            id: contactList.attributes.id,
+            contact_id: contactId
+          });
+        });
+
+        Promise.all(generatePromiseActions(contactListContactParams, createContactListContact))
+          .then(() => {
+            const campaignParams = {
+              name: 'Campaign 1',
+              title: 'Campaign 1 Title',
+              description: 'Campaign 1 description',
+              status: 'active',
+              contact_lists_id: contactList.attributes.id,
+              script_id: campaignScriptId
+            };
+
+            campaignsService.saveNewCampaign(campaignParams).then().catch(err => console.log(err));
```

Comment:
It seems that the campaign creation is dependent on the script creation as well but this step only really waits on contact list creation? This may be an implicit race condition that may occasionally fail if contact list creation finishes before script creation is complete. What do you think?

Reply:
Good point. Probably don't want to rely on contact and contact list and contact list contact creation taking longer than script and question and script question creation. I'll make it all one chain.